### PR TITLE
Routes cart #73

### PR DESCRIPTION
**Pull request by clocasto**
Large overhaul of all cart and order routes, DB handling, and factories. Should be able to manage carts and orders without error (except duplicate items in a cart -> not sure how to support that).

Also wrote barebones testing script for carts. All server testing except for this file is disabled because modifying any server file would cause the development database to drop.

### CODE REVIEW

**Review by nrod80 — APPROVED** [review]
this looks really solid, nice work. I'm going to approve, but I do have two quick things: Remind me what it means for a cart to be 'active' and what is the .addScope in the associations?

File: browser/js/factories/orders.factory.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+app.factory('CartFactory', function($http) {
```
Comment (curiousYi) [inline browser/js/factories/orders.factory.js]: I think this should be OrdersFactory?

**Review by curiousYi — COMMENTED** [review]
I think the factory in browser/js/factories/orders.factory.js is misnamed.

File: browser/js/factories/orders.factory.js (hunk 1 above)
Comment (clocasto) [reply browser/js/factories/orders.factory.js]: Good catch!

**Pull request merged**